From c623ca19f418c7f6d1d9dbf7ac6a80973c3ac7ab Mon Sep 17 00:00:00 2001
From: Dusty Mabe <dusty@dustymabe.com>
Date: Fri, 9 Feb 2024 22:19:35 -0500
Subject: [PATCH] mount/ostree.deployment: Fix ostree deployment call

We need to pass in the root of the ostree deployment which can
be the tree or the mount. Fixes e1cbf92
---
 mounts/org.osbuild.ostree.deployment | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mounts/org.osbuild.ostree.deployment b/mounts/org.osbuild.ostree.deployment
index 9637b5b2..c72dec6e 100755
--- a/mounts/org.osbuild.ostree.deployment
+++ b/mounts/org.osbuild.ostree.deployment
@@ -125,7 +125,6 @@ class OSTreeDeploymentMount(mounts.MountService):
         options = args["options"]
         source = options.get("source", "tree")
         deployment = options["deployment"]
-        osname, ref, serial = ostree.parse_deployment_option(tree, deployment)
 
         # The user could specify either the tree or mountroot as the
         # place where we want the deployment to be mounted.
@@ -134,6 +133,8 @@ class OSTreeDeploymentMount(mounts.MountService):
         else:
             target = tree
 
+        osname, ref, serial = ostree.parse_deployment_option(target, deployment)
+
         # create a private mountpoint for the target path, which is
         # needed in order to be able to move the deployment `root`
         # mountpoint here, which is contained inside tree, since
-- 
2.43.0

